server/src/main/java/com/thoughtworks/go/server/service/plugins/processor/elasticagent/ElasticAgentRequestProcessor.java

```diff
@@ -16,135 +16,56 @@
-        this.elasticAgentExtension = elasticAgentExtension;
-        this.agentConfigService = agentConfigService;
-        this.agentService = agentService;
-
-        registry.registerProcessorFor(PROCESS_DISABLE_AGENTS, this);
-        registry.registerProcessorFor(PROCESS_DELETE_AGENTS, this);
-        registry.registerProcessorFor(REQUEST_SERVER_LIST_AGENTS, this);
-    }
-
-    @Override
-    public GoApiResponse process(final GoPluginDescriptor pluginDescriptor, GoApiRequest goPluginApiRequest) {
-        switch (goPluginApiRequest.api()) {
-            case PROCESS_DISABLE_AGENTS:
-                Collection<AgentMetadata> agentsToDisable = elasticAgentExtension.getElasticAgentMessageConverter(goPluginApiRequest.apiVersion()).deleteAndDisableAgentRequestBody(goPluginApiRequest.requestBody());
-                if (agentsToDisable.isEmpty()) {
-                    return new DefaultGoApiResponse(200);
-                }
-                return processDisableAgent(pluginDescriptor, goPluginApiRequest);
-            case PROCESS_DELETE_AGENTS:
-                Collection<AgentMetadata> agentsToDelete = elasticAgentExtension.getElasticAgentMessageConverter(goPluginApiRequest.apiVersion()).deleteAndDisableAgentRequestBody(goPluginApiRequest.requestBody());
-                if (agentsToDelete.isEmpty()) {
-                    return new DefaultGoApiResponse(200);
-                }
-                return processDeleteAgent(pluginDescriptor, goPluginApiRequest);
-            case REQUEST_SERVER_LIST_AGENTS:
-                return processListAgents(pluginDescriptor, goPluginApiRequest);
-            default:
-                return DefaultGoApiResponse.error("Illegal api request");
-        }
-    }
-
-    private GoApiResponse processListAgents(GoPluginDescriptor pluginDescriptor, GoApiRequest goPluginApiRequest) {
-        LOGGER.debug("Listing agents for plugin {}", pluginDescriptor.id());
-        List<ElasticAgentMetadata> elasticAgents = agentService.allElasticAgents().get(pluginDescriptor.id());
-
-        Collection<AgentMetadata> metadata;
-        if (elasticAgents == null) {
-            metadata = new ArrayList<>();
-        } else {
-            metadata = elasticAgents.stream().map(new Function<ElasticAgentMetadata, AgentMetadata>() {
-                @Override
-                public AgentMetadata apply(ElasticAgentMetadata obj) {
-                    return toAgentMetadata(obj);
-                }
-            }).collect(Collectors.toList());
-        }
-
-        String responseBody = elasticAgentExtension.getElasticAgentMessageConverter(goPluginApiRequest.apiVersion()).listAgentsResponseBody(metadata);
-        return DefaultGoApiResponse.success(responseBody);
+    public ElasticAgentRequestProcessor(PluginRequestProcessorRegistry registry, AgentService agentService, AgentConfigService agentConfigService) {
+        this(registry, new ElasticAgentRequestProcessorV1(agentService, agentConfigService),
```

Comment:
I think this is incorrect. While the elastic agent extension version has changed over time, the processor contract hasn't changed at all. Why create three versions for this then?

Reply:
Our skeleton plugin uses same API constant to represent processor version and the extension version. That means the plugin that we have in gocd-contrib is already using "2.0" API version of the processor.